google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineApplicationInfoPanel.java

```diff
@@ -111,15 +120,32 @@ public boolean isApplicationValid() {
     return isApplicationValid;
   }
 
-  private void setMessage(String message, boolean isError) {
-    ApplicationManager.getApplication().invokeLater(() -> {
-      errorIcon.setVisible(false);
-      messageText.setText(HTML_OPEN_TAG + message + HTML_CLOSE_TAG);
-      messageText.setForeground(isError ? JBColor.red : JBColor.black);
-
+  private void setMessage(Runnable messagePrinter, boolean isError) {
+    ApplicationManager.getApplication().invokeAndWait(() -> {
+      errorIcon.setVisible(isError);
+      messagePrinter.run();
     }, ModalityState.stateForComponent(this));
   }
 
+  /**
+   * Prints a message that doesn't contain a hyperlink.
+   */
+  public void setMessage(String text, boolean isError) {
+    setMessage(() -> {
+      messageText.setText(text);
+      // HyperlinkLabels require that revalidate() be called after setText(), in order for text to
+      // actually show up. setHyperlinkText() calls revalidate() internally.
+      messageText.revalidate();
+    }, isError);
+  }
+
+  /**
+   * Prints a message with a hyperlink.
+   */
+  private void setMessage(String beforeLinkText, String linkText, String afterLinkText) {
+    setMessage(() -> messageText.setHyperlinkText(beforeLinkText, linkText, afterLinkText), true);
```

Comment:
An alternative would be to create three separate labels, two with the normal label on the sides and a hyperlink one in the middle. Though that's basically the same.